pkg/drivers/sqlite/sqlite.go

```diff
@@ -81,27 +63,181 @@ func NewVariant(ctx context.Context, driverName, dataSourceName string) (server.
 		time.Sleep(time.Second)
 	}
 	if err != nil {
-		return nil, nil, errors.Wrap(err, "setup db")
+		return nil, nil, errors.Wrap(err, "db table creation failed")
 	}
-	//if err := setup(dialect.DB); err != nil {
-	//	return nil, nil, errors.Wrap(err, "setup db")
-	//}
 
-	dialect.Migrate(context.Background())
 	if err := dialect.Prepare(); err != nil {
-		return nil, nil, err
+		return nil, nil, errors.Wrap(err, "query preparation failed")
 	}
 
 	return logstructured.New(sqllog.New(dialect)), dialect, nil
 }
 
-func setup(db *sql.DB) error {
-	for _, stmt := range schema {
-		_, err := db.Exec(stmt)
+// setup performs table setup, which may include creation of the Kine table if
+// it doesn't already exist, migrating key_value table contents to the Kine
+// table if the key_value table exists, all in a single database transaction.
+// changes are rolled back if an error occurs.
+func setup(dialect *generic.Generic) error {
+	ctx := context.Background()
+	txn, err := dialect.DB.BeginTx(ctx, nil)
+	if err != nil {
+		return err
+	}
+	defer txn.Rollback()
+
+	kineTableCount, _ := countTable(txn, context.Background(), "kine")
+	// Create Kine table if it doesn't already exist
+	if kineTableCount == 0 {
+		if err := createTable(txn); err != nil {
+			return err
+		}
+	}
+
+	if err := migration(txn, ctx, dialect); err != nil {
+		return errors.Wrap(err, "migration failed")
+	}
+	txn.Commit()
```

Comment:
`txn.Commit()` returns an error and needs to be checked... maybe `return txn.Commit()`?

Reply:
Thanks, good catch! Returning the result of `txn.Commit()` now.